### ISSUE TITLE
Load tf.keras weights until a specific layer?

### ISSUE THREAD

**Issue by innat** [tensorflow/models#9529]
Let's say, model `EfficientNet`. It has seven blocks in total. If we print out as follows: 

```python
import tensorflow as tf
func_model = tf.keras.applications.EfficientNetB0(input_shape=(224,224,3),
                                     include_top = True, weights=None)

for i, layer in enumerate(func_model.layers):
    print(i,'\t',layer.trainable,'\t  :',layer.name)
```

we'll get

```
0 	 True 	  : input_9
1 	 True 	  : rescaling_8
2 	 True 	  : normalization_8
3 	 True 	  : stem_conv_pad
...
9 	 True 	  : block1a_activation
10 	 True 	  : block1a_se_squeeze
11 	 True 	  : block1a_se_reshape
...
230 	 True 	  : block7a_se_expand
231 	 True 	  : block7a_se_excite
232 	 True 	  : block7a_project_conv
233 	 True 	  : block7a_project_bn
234 	 True 	  : top_conv
235 	 True 	  : top_bn
236 	 True 	  : top_activation
237 	 True 	  : avg_pool
238 	 True 	  : top_dropout
239 	 True 	  : predictions
```

Now, we can set `include_top = False`, and the layer from **237** to **239** will be removed and we can load no top pretrained weight. Now, my question is, what if I want to remove fully the `block_7*` and also at the same time want to load the pretrained weights from start to `block_6*`. How many ways can we do that? And which one is more precise and safe?

**Comment by innat** [comment]
**One Way**

If we want to exclude the last 10 layer 

```python
224 	 True 	  : block7a_dwconv
225 	 True 	  : block7a_bn
226 	 True 	  : block7a_activation

# ------------------ REMOVE ------------------
227 	 True 	  : block7a_se_squeeze
228 	 True 	  : block7a_se_reshape
229 	 True 	  : block7a_se_reduce
230 	 True 	  : block7a_se_expand
231 	 True 	  : block7a_se_excite
232 	 True 	  : block7a_project_conv
233 	 True 	  : block7a_project_bn
234 	 True 	  : top_conv
235 	 True 	  : top_bn
236 	 True 	  : top_activation
```

then we can do as follows: 

```python
func_model = tf.keras.applications.EfficientNetB0(input_shape=(224,224,3),
                                     include_top = False, weights='imagenet')

new_func_model = tf.keras.models.Model( inputs=func_model.inputs,
                                        outputs=func_model.layers[-11].get_output_at(0))
```

Fair enough. Any other way?

**Comment by innat** [comment]
**Another way**

By layer name, it seems more appropriate and easy to follow. Great. 
```python
for layer in func_model.layers:
        if layer.name == 'block6d_add':
            output = layer.output

i = tf.keras.models.Model(
    inputs=func_model.inputs, 
    outputs=output)
```

**Comment by ameyparanjape** [comment]
@innat This is interesting! What was the purpose of doing this? Did you want to reduce the model size? or any other reason? Also if you manipulate the model like this, did you observe any changes (good/bad) in your model performance both from the POV of inference time and accuracy?
